File: Java/2D Array(Matrix)/Divide Array Into Arrays With Max Difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int[][] ans =new int[nums.length/3][3];
        Arrays.sort(nums);
        for(int i=0;i<nums.length;i=i+3){
            if(nums[i+2]-nums[i]>k){
                return new int[0][0];
            }
            ans[i/3]=new int[]{nums[i],nums[i+1],nums[i+2]};
        }
        return ans;
    }
}

File: Python/2D Matrix/Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        n=len(nums)
        nums.sort()
        ans=[]
        for i in range(0,n,3):
            if nums[i+2]-nums[i]>k:
                return []
            else:
                ans.append([nums[i],nums[i+1],nums[i+2]])
        return ans

        
